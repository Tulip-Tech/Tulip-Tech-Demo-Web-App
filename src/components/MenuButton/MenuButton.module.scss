@use 'src/styles/variables';
@use 'src/styles/theme';
@use 'src/styles/mixins/responsive';
@use 'src/styles/mixins/utils';

//
// Navigation Button
// --------------------------------

.menuButton {
  position: relative;
  display: inline-flex;
  justify-content: inherit;
  align-items: center;
  width: 100%;
  min-height: 44px;
  padding: 0 variables.$base-spacing;
  color: variables.$white;
  font-family: var(--body-alt-font-family);
  font-weight: var(--body-font-weight-bold);
  font-size: 18px;
  text-align: center;
  text-decoration: none;
  background: transparent;
  outline: none;
  cursor: pointer;
  opacity: 0.7;
  transition: background 0.1s ease;

  &.small {
    padding: 0 variables.$base-spacing * 1.5;
    font-size: 16px;
    > .startIcon {
      margin-right: variables.$base-spacing * 1.5;
    }
  }

  @media (hover: hover) and (pointer: fine) {
    &:hover,
    &:active {
      background: rgba(variables.$white, 0.08);
      opacity: 1;
    }
  }
}

.active {
  background: rgba(variables.$white, 0.08);
  opacity: 1;
}

.startIcon {
  display: flex;
  align-items: center;
  height: 100%;
  margin-right: 12px;

  > svg {
    width: 25px;
    height: 25px;
    fill: currentColor;

    @include responsive.tablet-only() {
      width: 20px;
      height: 20px;
    }
  }
}
