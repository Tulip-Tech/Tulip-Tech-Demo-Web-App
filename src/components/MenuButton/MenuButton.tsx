import React from 'react';
import { NavLink } from 'react-router-dom';
import classNames from 'classnames';

import styles from './MenuButton.module.scss';

type Props = {
  label?: string;
  to?: string;
  onClick?: () => void;
  tabIndex?: number;
  active?: boolean;
  startIcon?: JSX.Element;
  small?: boolean;
};

const MenuButton: React.FC<Props> = ({ label, to, onClick, tabIndex = 0, active = false, startIcon, small = false }: Props) => {
  const icon = startIcon ? <div className={styles.startIcon}>{startIcon}</div> : null;
  const getClassName = (isActive: boolean) => classNames(styles.menuButton, { [styles.small]: small }, { [styles.active]: isActive });

  if (to) {
    return (
      <NavLink aria-label={label} className={({ isActive }) => getClassName(isActive || active)} onClick={onClick} to={to} tabIndex={tabIndex} end>
        {icon}
        <span className={styles.label}>{label}</span>
      </NavLink>
    );
  }

  return (
    <div aria-label={label} className={getClassName(active)} onClick={onClick} tabIndex={tabIndex}>
      {icon}
      <span className={styles.label}>{label}</span>
    </div>
  );
};

export default MenuButton;
