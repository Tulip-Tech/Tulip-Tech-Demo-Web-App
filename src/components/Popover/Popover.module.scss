@use 'src/styles/variables';
@use 'src/styles/theme';

.popover {
  position: absolute;
  top: 55px;
  right: 5px;
  z-index: variables.$header-z-index;
  width: 250px;
  overflow: hidden;
  border-radius: 5px 5px;
  > div {
    z-index: variables.$header-z-index;
  }
}
