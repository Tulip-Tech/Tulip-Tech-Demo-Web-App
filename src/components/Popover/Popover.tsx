import React, { ReactNode } from 'react';
import classNames from 'classnames';

import styles from './Popover.module.scss';

import DetectOutsideClick from '#components/DetectOutsideClick/DetectOutsideClick';
import Slide from '#components/Animation/Slide/Slide';

type Props = {
  children: ReactNode;
  isOpen: boolean;
  onClose: () => void;
};

const Popover: React.FC<Props> = ({ children, isOpen, onClose }: Props) => {
  return (
    <Slide open={isOpen} duration={250} direction="right">
      <DetectOutsideClick callback={onClose}>
        <div className={classNames(styles.popover)}>{children}</div>
      </DetectOutsideClick>
    </Slide>
  );
};

export default Popover;
