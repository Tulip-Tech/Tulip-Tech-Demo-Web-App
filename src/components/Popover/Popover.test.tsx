import React from 'react';

import Popover from './Popover';

import { renderWithRouter } from '#test/testUtils';
import MenuButton from '#components/MenuButton/MenuButton';

describe('<Popover>', () => {
  test('renders and matches snapshot', () => {
    const menuItems = [<MenuButton key="key" label="Home" to="/" />];

    const { container } = renderWithRouter(
      <Popover isOpen={true} onClose={vi.fn()}>
        {menuItems}
      </Popover>,
    );

    expect(container).toMatchSnapshot();
  });
});
