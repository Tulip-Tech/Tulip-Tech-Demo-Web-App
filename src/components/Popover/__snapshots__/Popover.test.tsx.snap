// Vitest Snapshot v1

exports[`<Popover> > renders and matches snapshot 1`] = `
<div>
  <div
    style="transition: transform 0.25s ease, opacity 0.25s ease; transform: translate(15px, 0); opacity: 0; z-index: 15;"
  >
    <div
      class="_popover_9be12f"
    >
      <a
        aria-current="page"
        aria-label="Home"
        class="_menuButton_91706b _active_91706b"
        href="/"
        tabindex="0"
      >
        <span>
          Home
        </span>
      </a>
    </div>
  </div>
</div>
`;
